src/dotnet/ShellShim/OsxBashEnvironmentPath.cs

```diff
@@ -10,7 +10,7 @@
 
 namespace Microsoft.DotNet.ShellShim
 {
-    internal class OSXEnvironmentPath : IEnvironmentPath
+    internal class OsxBashEnvironmentPath : IEnvironmentPath
```

Comment:
Do we really need two types to implement a slightly different message?  Can't we keep this as-is and change `PrintAddPathInstructioNifPathDoesNotExist` to:

```c#
public void PrintAddPathInstructionIfPathDoesNotExist()
{
    if (!PackageExecutablePathExists())
    {
        if (_fileSystem.Exists(DotnetCliToolsPathsDPath))
        {
            _reporter.WriteLine(
                CommonLocalizableStrings.EnvironmentPathOSXNeedReopen);
        }
        else
        {
            // similar to https://code.visualstudio.com/docs/setup/mac
            _reporter.WriteLine(
                string.Format(
                    new ZshDetector(_environmentProvider).IsZshTheUsersShell() ?
                        CommonLocalizableStrings.EnvironmentPathOSXZshManualInstructions :
                        CommonLocalizableStrings.EnvironmentPathOSXBashManualInstructions,
                    _packageExecutablePath.Path));
        }
    }
}
```

Reply:
I assume macOS Catalina will keep paths.d working for zsh, right?

Reply:
There are more difference -- `PackageExecutablePathExists` in bash impl consider `~/.dotnet/tools` be good and skip printing. But it is not the case in zsh. Only full path `/home/myuser/.dotnet/tools` will work on zsh.

Also bash has interface `IEnvironmentPath` which as extra `AddPackageExecutablePathToUserPath` method. That does not exist in zsh (we can only print instruction).

So I think it is easier to reason about it with different impl

Reply:
Ah good point, I forgot that we put `~` on the PATH for bash.

Ok, let's keep it two different implementations.

Reply:
> I assume macOS Catalina will keep paths.d working for zsh, right?

There is no paths.d on macOS only path_helper.
and `path_helper` will not expand `~` like paths.d do. So it end up literal `~/.dotnet/tools` on PATH. which is good for bash but not for zsh